File: backend/backend/Areas/Communication/Models/Meeting.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using backend.Areas.Identity.Models;
using backend.Areas.Main.Models;

namespace backend.Areas.Communication.Models;

public class Meeting
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    
    [StringLength(200)]
    public string Title { get; set; }

    [StringLength(2000)]
    public string? Description { get; set; }
    
    public DateTime StartTime { get; set; }
    
    public DateTime EndTime { get; set; }

    public bool IsOnline { get; set; } = true;

    [StringLength(500)]
    public string? Location { get; set; }

    [StringLength(500)]
    public string? MeetingLink { get; set; }
    
    public string OrganizerId { get; set; }

    [ForeignKey(nameof(OrganizerId))]
    public User Organizer { get; set; }

    // Participants - many-to-many relationship with User
    public ICollection<UserMeeting> UserMeetings { get; set; }

    // Related Lead (optional)
    public int? LeadId { get; set; }

    [ForeignKey(nameof(LeadId))]
    public Lead? Lead { get; set; }

    // Related Contact (optional)
    public int? ContactId { get; set; }

    [ForeignKey(nameof(ContactId))]
    public Contact? Contact { get; set; }

    public DateTime DateCreated { get; set; } = DateTime.UtcNow;
}

File: frontend/lib/models/meeting.dart

import 'package:frontend/models/user_meeting.dart';
import 'package:json_annotation/json_annotation.dart';

@JsonSerializable()
class Meeting {
  int? id;
  String? title;
  String? description;
  DateTime? startTime;
  DateTime? endTime;
  bool? isOnline;
  String? location;
  String? meetingLink;
  String? organizerId;
  String? dateCreated;
  int? leadId;
  int? contactId;
  List<UserMeeting>? userMeetings;

  Meeting({
    this.id,
    this.title,
    this.description,
    this.startTime,
    this.endTime,
    this.isOnline = true,
    this.location,
    this.meetingLink,
    this.organizerId,
    this.leadId,
    this.contactId,
    this.dateCreated,
    this.userMeetings
  });

  factory Meeting.fromJson(Map<String, dynamic> json) => Meeting(
    id: json['id'],
    title: json['title'],
    description: json['description'],
    startTime: json['startTime'] != null ? DateTime.parse(json['startTime']) : null,
    endTime: json['endTime'] != null ? DateTime.parse(json['endTime']) : null,
    isOnline: json['isOnline'],
    location: json['location'],
    meetingLink: json['meetingLink'],
    organizerId: json['organizerId'],
    leadId: json['leadId'],
    contactId: json['contactId'],
    dateCreated: json['dateCreated'],
    userMeetings: (json['userMeetings'] as List?)?.map((e) => UserMeeting.fromJson(e)).toList(),
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'title': title,
    'description': description,
    'startTime': startTime?.toIso8601String(),
    'endTime': endTime?.toIso8601String(),
    'isOnline': isOnline,
    'location': location,
    'meetingLink': meetingLink,
    'organizerId': organizerId,
    'leadId': leadId,
    'contactId': contactId,
    'dateCreated': dateCreated,
    'userMeetings': userMeetings?.map((e) => e.toJson()).toList(),
  };
}